File: src/TickerQ.Utilities/Enums/RunCondition.cs
namespace TickerQ.Utilities.Enums
{
    /// <summary>
    /// Defines under what conditions a dependent ticker/job should run
    /// based on the final status of its parent.
    /// </summary>
    public enum RunCondition
    {
        /// <summary>
        /// Run only if the parent completed successfully
        /// (i.e., status is Done or DueDone).
        /// </summary>
        OnSuccess,

        /// <summary>
        /// Run only if the parent failed.
        /// </summary>
        OnFailure,

        /// <summary>
        /// Run only if the parent was cancelled by the user/system.
        /// </summary>
        OnCancelled,

        /// <summary>
        /// Run if the parent failed or was cancelled.
        /// </summary>
        OnFailureOrCancelled,

        /// <summary>
        /// Run after the parent reaches any terminal state
        /// except skipped (Done, DueDone, Failed, Cancelled).
        /// This is like "finally" semantics but excludes Skipped.
        /// </summary>
        OnAnyCompletedStatus,
        
        /// <summary>
        /// Run in parallel with the parent (when parent status is InProgress).
        /// </summary>
        InProgress
    }
}

File: src/TickerQ.Dashboard/wwwroot/src/lib/cron/run-condition.ts
export type RunCondition =
  | "OnSuccess"
  | "OnFailure"
  | "OnCancelled"
  | "OnFailureOrCancelled"
  | "OnAnyCompletedStatus"
  | "InProgress";

export interface RunConditionConfig {
  label: string;
  text: string;
  bg: string;
  cssVar: string;
  description: string;
}

export const RUN_CONDITION_CONFIG: Record<RunCondition, RunConditionConfig> = {
  OnSuccess:            { label: "On Success",     text: "text-status-healthy",    bg: "bg-status-healthy/10",    cssVar: "var(--status-healthy)",    description: "Run when parent completes with Done or DueDone" },
  OnFailure:            { label: "On Failure",     text: "text-status-error",      bg: "bg-status-error/10",      cssVar: "var(--status-error)",      description: "Run when parent fails" },
  OnCancelled:          { label: "On Cancelled",   text: "text-status-cancelled",  bg: "bg-status-cancelled/10",  cssVar: "var(--status-cancelled)",  description: "Run when parent is cancelled" },
  OnFailureOrCancelled: { label: "On Fail/Cancel", text: "text-status-warning",    bg: "bg-status-warning/10",    cssVar: "var(--status-warning)",    description: "Run when parent fails or is cancelled" },
  OnAnyCompletedStatus: { label: "On Any",         text: "text-status-info",       bg: "bg-status-info/10",       cssVar: "var(--status-info)",       description: "Run on any terminal status except Skipped" },
  InProgress:           { label: "Parallel",       text: "text-status-skipped",    bg: "bg-status-skipped/10",    cssVar: "var(--status-skipped)",    description: "Run concurrently with parent" },
};

export const RUN_CONDITION_OPTIONS: RunCondition[] = [
  "OnSuccess",
  "OnFailure",
  "OnCancelled",
  "OnFailureOrCancelled",
  "OnAnyCompletedStatus",
  "InProgress",
];

export function runConditionHsl(rc?: RunCondition): string {
  if (!rc) return "hsl(var(--border))";
  return `hsl(${RUN_CONDITION_CONFIG[rc].cssVar})`;
}
